cmd/ctr/commands/resolver.go

```diff
@@ -107,3 +113,30 @@ func GetResolver(ctx gocontext.Context, clicontext *cli.Context) (remotes.Resolv
 
 	return docker.NewResolver(options), nil
 }
+
+// loadCertsDir loads certs from certsDir like "/etc/docker/certs.d" .
+func loadCertsDir(config *tls.Config, certsDir string) {
```

Comment:
The could be better done using the new `Hosts` option on resolver to avoid reading all files. Let me help out get that updated, that is the ultimate goal for how this would be used and will be used by Docker.

Reply:
ping

Reply:
Will https://github.com/containerd/containerd/pull/4103 solve your concern?